tests/integration_tests/style/test_rust.py

```diff
@@ -21,40 +24,47 @@ def test_rust_style():
     assert "Diff in" not in stdout
 
 
-def test_ensure_mod_tests():
+def test_ensure_mod_tests(monkeypatch):
     """
     Check that files containing unit tests have a 'tests' module defined.
 
     @type: style
     """
-    # List all source files containing rust #[test] attribute,
-    # (excluding generated files and integration test directories).
-    # Take the list and check each file contains 'mod tests {', output file
-    # name if it doesn't.
-    cmd = (
-        "/bin/bash "
-        "-c "
-        '"grep '
-        "--files-without-match "
-        "'mod tests {' "
-        "\\$(grep "
-        "--files-with-matches "
-        "--recursive "
-        "--exclude-dir=src/*_gen/* "
-        "'\\#\\[test\\]' ../src/*/src)\" "
-        '| grep -v "../src/io_uring/src/bindings.rs"'
-    )
-
-    # The outer grep returns 0 even if it finds files without the match, so we
-    # ignore the return code.
-    result = utils.run_cmd(cmd, no_shell=False, ignore_return_code=True)
-
-    stdout = result.stdout.strip()
-
-    error_msg = (
-        f'Tests found in files without a "tests" module:\n {stdout} '
-        "To ensure code coverage is reported correctly, please check that "
-        'your tests are in a module named "tests".'
-    )
-
-    assert not stdout, error_msg
+    excluding = [
+        "_gen/",
+        "/tests/",
+        "/test_utils",
+        "build/",
+        "src/io_uring/src/bindings.rs",
+    ]
+
+    def cmd(string):
+        cmd = shlex.split(string)
+        process = subprocess.run(cmd, capture_output=True, check=True)
+        return process.stdout.decode("utf-8").split("\n")
+
+    def disjoint(one, two):
+        return list(set(one) - set(two))
+
+    # Re-direct to repository root.
+    monkeypatch.chdir("..")
+
+    # Install ripgrep
+    subprocess.run(["cargo", "install", "ripgrep"], capture_output=True, check=True)
```

Comment:
1. This wouldn't significantly impact test performance in an environment where the tool was already installed.
2. In retrospect I agree, I can remove the install (is useful for now to see CI result without needing to update docker container) (I think you meant `but it would *not* be good to add more`)

Reply:
This no longer applies (as ripgrep has been removed).